### PR TITLE
Data & view binding

### PR DESCRIPTION
**Pull request by mustafaAnis**
Databinding implemented on:

- Ayat recycle adapter
- surah recycle adapter
- flashes recycle adapter

### CODE REVIEW

**Comment by mustafaAnis** [comment]
@AsadLeo1995

File: app/src/main/java/com/example/androidapplicationtask/AyaatRecyclerAdapter.kt [hunk 1]
```diff
@@ -2,61 +2,52 @@ package com.example.androidapplicationtask
...
+import androidx.databinding.DataBindingUtil
 import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import com.example.androidapplicationtask.databinding.AyatItemViewBinding
 
 class AyaatRecyclerAdapter(
     val ayaat: List<AyaatData>,
-    val onItemClickListenerForAyaat: OnItemClickListenerForAyaat,
+    val listner: OnItemClickListenerForAyaat,
```
Comment (AsadLeo1995) [inline app/src/main/java/com/example/androidapplicationtask/AyaatRecyclerAdapter.kt:19]: give proper name for listener i.e 'OnItemClickListene'

File: app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        if (flag != 1) {
+            this.setImageResource(R.drawable.holo_bookmark)
+        } else {
+            this.setImageResource(R.drawable.solid_bookmark)
+        }
+
+
+}
+@BindingAdapter("setFlashes")
+fun ImageView.setFlashes(flashes : Post){
```
Comment (AsadLeo1995) [inline app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt:21]: method argument should be url not post object

Comment (AsadLeo1995) [inline app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt:21]: method name should be `setImage`

File: app/src/main/java/com/example/androidapplicationtask/FlashesRecyclerAdapter.kt [hunk 3]
```diff
@@ -7,23 +7,22 @@ import android.view.View
...
 import android.widget.ImageView
 import android.widget.Toast
+import androidx.databinding.DataBindingUtil
 import androidx.recyclerview.widget.RecyclerView
+import com.example.androidapplicationtask.databinding.FlashesListViewBinding
 import com.squareup.picasso.Picasso
 
 class FlashesRecyclerAdapter(val flashes: List<Post>, val context: Context) :
     RecyclerView.Adapter<FlashesRecyclerAdapter.FlahsesViewHolder>() {
     var url = "https://islam360.app/islam360/posts/image/urdu/"
```
Comment (AsadLeo1995) [inline app/src/main/java/com/example/androidapplicationtask/FlashesRecyclerAdapter.kt:17]: if there is no usage for this url kindly remove it

File: app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package com.example.androidapplicationtask
+
+import android.widget.ImageView
+import androidx.databinding.BindingAdapter
+import com.squareup.picasso.Picasso
+
+
+@BindingAdapter("setBookmarkImage")
+fun ImageView.setBookmarkImage(flag: Int){
```
Comment (AsadLeo1995) [inline app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt:9]: change data type of flag from Int to Boolean

File: app/src/main/java/com/example/androidapplicationtask/OnItemClickListenerForSurah.kt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package com.example.androidapplicationtask
+
+interface OnItemClickListenerForSurah {
+
+    fun onItemClickSurahCard(position: Int)
```
Comment (AsadLeo1995) [inline app/src/main/java/com/example/androidapplicationtask/OnItemClickListenerForSurah.kt:5]: rename this method to `onSurahClick`

**Review by AsadLeo1995 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/androidapplicationtask/AyaatRecyclerAdapter.kt (hunk 1 above)
Comment (mustafaAnis) [reply app/src/main/java/com/example/androidapplicationtask/AyaatRecyclerAdapter.kt]: done

File: app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt (hunk 2 above)
Comment (mustafaAnis) [reply app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt]: done

Comment (mustafaAnis) [reply app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt]: done

File: app/src/main/java/com/example/androidapplicationtask/FlashesRecyclerAdapter.kt (hunk 3 above)
Comment (mustafaAnis) [reply app/src/main/java/com/example/androidapplicationtask/FlashesRecyclerAdapter.kt:12]: done

File: app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt (hunk 4 above)
Comment (mustafaAnis) [reply app/src/main/java/com/example/androidapplicationtask/BindingAdapter.kt:9]: done

File: app/src/main/java/com/example/androidapplicationtask/OnItemClickListenerForSurah.kt (hunk 5 above)
Comment (mustafaAnis) [reply app/src/main/java/com/example/androidapplicationtask/OnItemClickListenerForSurah.kt]: done

**Pull request merged**